make_single_header.py

```diff
@@ -26,27 +35,50 @@ def process_file(file_path, out_lines=[], front_matter_lines=[], back_matter_lin
                 if inc_path not in processed_files:
                     if inc_path is not None:
                         processed_files += [inc_path]
-                        process_file(inc_path, out_lines, front_matter_lines, back_matter_lines, processed_files)
+                        process_file(
+                            inc_path,
+                            out_lines,
+                            front_matter_lines,
+                            back_matter_lines,
+                            processed_files,
+                        )
                     else:
-                        # assume it's a system header; add it to the front matter just to be clean
-                        front_matter_lines += [line]
+                        # assume it's a system header
+                        if inc_name not in system_headers:
+                            out_lines += [line]
+                        else:
+                            system_headers += [inc_name]
```

Comment:
I don't quite understand this. In the else branch aren't you adding inc_name to system_headers again? Since the condition to hit the else branch is that inc_name is already in system_headers?

Reply:
Oops, my bad, this block is a mistake. We should just add it to out_lines, without any check. otherwise, there will be conflicted with eg: [config.hpp](https://github.com/kokkos/mdspan/blob/stable/include/experimental/__p0009_bits/config.hpp#L22-L27) and [macros.hpp](https://github.com/kokkos/mdspan/blob/stable/include/experimental/__p0009_bits/macros.hpp#L21). When compiler can find `version`, `type_traits` in macros.hpp can't be add if ignore system_header which already exist.